tests/system-tests/rdscore/internal/rdscorecommon/sriov-rootless-dpdk.go

```diff
@@ -0,0 +1,861 @@
+
+		return nil, fmt.Errorf("failed to build a links object list for %q due to %w",
+			linksRawInfo, err)
+	}
+
+	for _, linkInfo := range linksInfoList {
+		linksInfoMap[linkInfo.Ifname] = linkInfo.GetRxByte()
+	}
+
+	return linksInfoMap, nil
+}
+
+func checkRxOnly(out string) bool {
+	lines := strings.Split(out, "\n")
+	for index, line := range lines {
+		if strings.Contains(line, "NIC statistics for port") {
+			if len(lines[index+1]) < 3 {
+				glog.V(90).Info("Fail: line list contains less than 3 elements")
+
+				return false
+			}
+
+			if len(lines) > index && getNumberOfPackets(lines[index+1], "RX") > 0 {
+				return true
+			}
+		}
+	}
+
+	return false
+}
+
+func getNumberOfPackets(line, firstFieldSubstr string) int {
+	splitLine := strings.Fields(line)
+
+	if !strings.Contains(splitLine[0], firstFieldSubstr) {
+		glog.V(90).Infof("Failed to find expected substring %s", firstFieldSubstr)
+
+		return 0
+	}
+
+	if len(splitLine) != 6 {
+		glog.V(90).Info("the slice doesn't contain 6 elements")
+
+		return 0
+	}
+
+	numberOfPackets, err := strconv.Atoi(splitLine[1])
+
+	if err != nil {
+		glog.V(90).Infof("failed to convert string to integer %s", err)
+
+		return 0
+	}
+
+	return numberOfPackets
+}
+
+func defineTestServerPmdCmd(ethPeer, pciAddress, txIPs string) []string {
+	baseCmd := fmt.Sprintf("dpdk-testpmd -a %s -- --forward-mode txonly --eth-peer=0,%s", pciAddress, ethPeer)
+	if txIPs != "" {
+		baseCmd += fmt.Sprintf(" --tx-ip=%s", txIPs)
+	}
+
+	baseCmd += " --stats-period 5"
+
+	return []string{"/bin/bash", "-c", baseCmd}
+}
+
+func defineTestPmdCmd(interfaceName string, pciAddress string) string {
+	return fmt.Sprintf("timeout -s SIGKILL 20 dpdk-testpmd "+
+		"--vdev=virtio_user0,path=/dev/vhost-net,queues=2,queue_size=1024,iface=%s "+
+		"-a %s -- --stats-period 5", interfaceName, pciAddress)
+}
+
+func checkRxOutputRateForInterfaces(
+	clientPod *pod.Builder,
+	origInterfaceTrafficRateMap,
+	interfaceTrafficRateMap map[string]int) error {
+	for interfaceName, TrafficRate := range interfaceTrafficRateMap {
+		originalRate := origInterfaceTrafficRateMap[interfaceName]
+
+		glog.V(100).Infof("Original Rate for the interface %s: %d", interfaceName, originalRate)
+
+		currentRate, err := getLinkRx(clientPod, interfaceName)
+		if err != nil {
+			glog.V(100).Infof("Failed to collect link %s info from pod %s in namespace %s with error %v",
+				interfaceName, clientPod.Definition.Name, clientPod.Definition.Namespace, err)
+
+			return fmt.Errorf("failed to get link %s info from pod %s in namespace %s with error %w",
+				interfaceName, clientPod.Definition.Name, clientPod.Definition.Namespace, err)
+		}
+
+		glog.V(100).Infof("Current Rate for the interface %s: %d", interfaceName, currentRate)
+
+		currentRate -= originalRate
+
+		glog.V(100).Infof("Current run Rate for the interface %s: %d", interfaceName, currentRate)
+
+		if TrafficRate == 5000 && currentRate > TrafficRate || TrafficRate != 5000 && currentRate < TrafficRate {
```

Comment:
why such a special treatment for  `5000`?

Reply:
this is maximum acceptable rate for the unrelevant interface

Reply:
Ack, thanks.  In this case - shall it be moved to some kind of _constant_ ?

Reply:
ah, probably it was already done, I just forget to replace to the constant in this specific place :)

Reply:
fixed